src/libasr/asr_utils.h

```diff
@@ -38,49 +38,6 @@ static inline ASR::ttype_t* TYPE(const ASR::asr_t *f)
     return ASR::down_cast<ASR::ttype_t>(f);
 }
 
-static inline char *symbol_name(const ASR::symbol_t *f)
-{
-    switch (f->type) {
-        case ASR::symbolType::Program: {
-            return ASR::down_cast<ASR::Program_t>(f)->m_name;
-        }
-        case ASR::symbolType::Module: {
-            return ASR::down_cast<ASR::Module_t>(f)->m_name;
-        }
-        case ASR::symbolType::Subroutine: {
-            return ASR::down_cast<ASR::Subroutine_t>(f)->m_name;
-        }
-        case ASR::symbolType::Function: {
-            return ASR::down_cast<ASR::Function_t>(f)->m_name;
-        }
-        case ASR::symbolType::GenericProcedure: {
-            return ASR::down_cast<ASR::GenericProcedure_t>(f)->m_name;
-        }
-        case ASR::symbolType::DerivedType: {
-            return ASR::down_cast<ASR::DerivedType_t>(f)->m_name;
-        }
-        case ASR::symbolType::Variable: {
-            return ASR::down_cast<ASR::Variable_t>(f)->m_name;
-        }
-        case ASR::symbolType::ExternalSymbol: {
-            return ASR::down_cast<ASR::ExternalSymbol_t>(f)->m_name;
-        }
-        case ASR::symbolType::ClassProcedure: {
-            return ASR::down_cast<ASR::ClassProcedure_t>(f)->m_name;
-        }
-        case ASR::symbolType::CustomOperator: {
-            return ASR::down_cast<ASR::CustomOperator_t>(f)->m_name;
-        }
-        case ASR::symbolType::AssociateBlock: {
-            return ASR::down_cast<ASR::AssociateBlock_t>(f)->m_name;
-        }
-        case ASR::symbolType::Block: {
-            return ASR::down_cast<ASR::Block_t>(f)->m_name;
-        }
-        default : throw LFortranException("Not implemented");
-    }
-}
-
```

Comment:
Why is this removed? This is needed.

Reply:
I guess this comment is about the function `static inline char *symbol_name(const ASR::symbol_t *f)` (from [here](https://github.com/lcompilers/lpython/pull/730/files/c248ae1b6112b2a0436adab2327b5742de4ab586#diff-da3dce35379cdaaf286c90ae38fe3d27efda9a299f60c682dacc97ad1eca5ce7L41)).

It is defined below [here](https://github.com/lcompilers/lpython/pull/730/files/c248ae1b6112b2a0436adab2327b5742de4ab586#diff-da3dce35379cdaaf286c90ae38fe3d27efda9a299f60c682dacc97ad1eca5ce7R193). It is defined just above `static inline SymbolTable *symbol_parent_symtab(const ASR::symbol_t *f)` ([here](https://github.com/lcompilers/lpython/pull/730/files/c248ae1b6112b2a0436adab2327b5742de4ab586#diff-da3dce35379cdaaf286c90ae38fe3d27efda9a299f60c682dacc97ad1eca5ce7L226)) and this convention is taken from `LFortran`. (In `LFortran`, the function `static inline char *symbol_name(const ASR::symbol_t *f)` was defined just above `static inline SymbolTable *symbol_parent_symtab(const ASR::symbol_t *f)`, so the same convention is followed here).

Reply:
Yes. That's why I minimized my comment after noticing that later.